core/backend/src/IModelDb.ts

```diff
@@ -3075,10 +3080,19 @@ export class SnapshotDb extends IModelDb {
   }
 
   /**
-   * Open a Checkpoint directly from its cloud container.
+   * Open a Checkpoint directly from its cloud container in a "lazy" manner, allowing data to be fetched as needed.
+   * This method can optionally be combined with a prefetch to download all blocks of an iModel if desired.
+   * Prefetching can improve performance when a large portion of the database will be needed, performance would be better the more you needed to read if you happened to prefetch.
+   * But use it smartly to avoid unnecessary data transfer and storage.
+   * If only a small portion of data is needed, it's more efficient to rely solely on lazy opening.
+   * @param args The arguments required to open the checkpoint.
+   * @param returns A promise that resolves with an instance of SnapshotDb.
    * @beta
    */
   public static async openCheckpoint(args: OpenCheckpointArgs): Promise<SnapshotDb> {
+    if (args.prefetch) {
+      IModelHost.appWorkspace.settings.addDictionary("prefetch", SettingsPriority.application, {"Checkpoints/prefetch": true});
+    }
```

Comment:
No, you shouldn't set create a setting. Add something to the argument to `V2CheckpointManager.attach`

Reply:
I was unsure about this, part of me says we should activate a prefetch in the same way that RPC does (through settings at the moment). What do you think about starting a prefetch manually in the `private static async attachAndOpenCheckpoint` function? We'd add a line CloudSqlite.startCloudPrefetch, we already have the dbName and container in that function after it returns from the attach so we have all that we need.

Reply:
yes, but then you'd have to reproduce all the minRequest/maxrequest and logging logic. I will pull this branch and add an outline of how i think it should work.

Reply:
We should also allow for CheckpointConnection.ts: openRemote to pass a prefetch property, no? That is the main way that a desktop app would actually be able to use this?

Reply:
that would be nice, but stupidly that function's signature can't be changed (because the last argument is optional.) I guess we could deprecate the current signature and add a new one with named arguments.

Reply:
actually, that method is only used from RPC where the backend decides whether or not to prefetch, so it shouldn't change.

Reply:
nope, wrong again. That function is also used for IPC.  I'm not sure what to do. I think it's probably fine to just have it use the default (doPrefetch=true).

Reply:
> that would be nice, but stupidly that function's signature can't be changed (because the last argument is optional.) I guess we could deprecate the current signature and add a new one with named arguments.

Yes I also was struggling with that, feels like openRemote should take a single arg which is an interface as opposed to a list of args. Is it really a breaking change to add a 4th optional arg? 

Current usage of openRemote would be 
openRemote(itwinid, imodelid)
openRemote(itwinid, imodelid, imodelversion.latest())

After the change I believe all of the following would also be acceptable:
openRemote(itwinid,imodelid, imodelversion.latest())  
openRemote(itwinid,imodelid) 
openRemote(itwinid, imodelid, imodelversion.latest(), false)

Reply:
That works